src/event_key.rs

```diff
@@ -2,10 +2,42 @@
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
+extern crate libc;
 use glib::translate::*;
+use std::ffi::CStr;
 
 #[derive(Clone, Debug)]
 pub struct EventKey(::Event);
 
 event_wrapper!(EventKey, GdkEventKey);
 event_subtype!(EventKey, KeyPress | KeyRelease);
+
+impl EventKey {
+    pub fn get_time(&self) -> u32 {
+        self.as_ref().time
+    }
+
+    pub fn get_state(&self) -> ::ModifierType {
+        self.as_ref().state
+    }
+
+    pub fn get_keyval(&self) -> u32 {
+        self.as_ref().keyval
+    }
+
+    pub fn get_length(&self) -> u32 {
+        let length = self.as_ref().length;
+        assert!(length >= 0, "Unexpected negative value");
+        length as u32
+    }
+
+    pub fn get_hardware_keycode(&self) -> ::enums::key::Key {
```

Comment:
I think this and `get_keyval` need to be the other way around: `enums::key` is about keyvals and `Key` is `u32`, while `hardware_keycode` is `u16`.